Change network partitioning method (DSN),

shamil-gadelshin opened the pull request:
This PR changes the network partitioning method from protocol name prefixes to protocol version in the `Identify` protocol messages.

### Code contributor checklist:
* [x] I have read, understood and followed [contributing guide](https://github.com/subspace/subspace/blob/main/CONTRIBUTING.md)

shamil-gadelshin committed:
Change network partition method.

nazar-pc commented on crates/subspace-networking/src/create.rs:
```diff
@@ -295,7 +290,8 @@ where
         });
 
-        let identify = IdentifyConfig::new("ipfs/0.1.0".to_string(), keypair.public());
+        let protocol_version = SUBSPACE_PROTOCOL_VERSION_PREFIX.to_string() + &protocol_version;
```
You really should have used `format!()`

shamil-gadelshin replied:
That was my first option but I expected an opposite comment based on the previous PRs 🤷

shamil-gadelshin committed:
Apply small fixes.

- fix build error (dsn_protocol_prefix -> dsn_protocol_version)
- remove SUBSPACE_PROTOCOL_VERSION_PREFIX const

nazar-pc reviewed (approved):
I think clippy will complain, but if not - fine
